src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -5,19 +5,126 @@ import { Button, Group, Stack, Text } from '@mantine/core'
 import { notifications } from '@mantine/notifications'
 import { CancelButton } from '@/components/cancel-button'
 import { useForm } from '@mantine/form'
-import { StudyProposalFormValues, studyProposalSchema, zodResolver } from './study-proposal-schema'
+import { StudyProposalFormValues, studyProposalSchema } from './study-proposal-schema'
 import { StudyProposalForm } from './study-proposal-form'
 import { UploadStudyJobCode } from './upload-study-job-code'
 import { useMutation } from '@tanstack/react-query'
 import { onCreateStudyAction } from './actions'
 import { useRouter } from 'next/navigation'
+import { StudyDocumentType } from '@/lib/types'
+import { zodResolver } from 'mantine-form-zod-resolver'
+import { CodeReviewManifest } from '@/lib/code-manifest'
+import { PresignedPost } from '@aws-sdk/s3-presigned-post'
+import { signedUrlForCodeUploadAction, signedUrlForStudyFileUploadAction } from '@/server/actions/s3.actions'
 
-export const StudyProposal: React.FC<{ memberId: string }> = ({ memberId }) => {
+// TODO @nathan, should we talk about local vs s3 storage?
+//  Could we just use s3 locally and not have to switch on USING_S3 to simplify?
+async function uploadFile(file: File, upload: PresignedPost) {
+    const body = new FormData()
+    for (const [key, value] of Object.entries(upload.fields)) {
+        body.append(key, value)
+    }
+    body.append('file', file)
+
+    const response = await fetch(upload.url, {
+        method: 'POST',
+        body,
+    })
+
+    if (!response.ok) {
+        notifications.show({
+            color: 'red',
+            title: 'failed to upload file',
+            message: await response.text(),
+        })
+    }
+
+    return response.ok
+}
+
+async function uploadCodeFiles(files: File[], upload: PresignedPost, studyJobId: string) {
+    const manifest = new CodeReviewManifest(studyJobId, 'r')
+    const body = new FormData()
+    for (const [key, value] of Object.entries(upload.fields)) {
+        body.append(key, value)
+    }
+    for (const codeFile of files) {
+        manifest.files.push(codeFile)
+        body.append('file', codeFile)
+    }
+
+    const manifestFile = new File([manifest.asJSON], 'manifest.json', { type: 'application/json' })
+    body.append(manifestFile.name, manifestFile)
+
+    await fetch(upload.url, {
+        method: 'POST',
+        body: body,
+    })
+}
```

Comment:
there doesn't seem to be any error handling here.  what happens if the upload does not succeed?